### ISSUE TITLE
Release 1.12.1

### ISSUE THREAD

**Issue by squakez** [apache/camel-k#4396]
Release tracker

**Comment by squakez** [comment]
@oscerd I will start next week the process to release. I may need your help, I'll ping you in case, is it okey?

**Comment by squakez** [comment]
The only one that may still require some attention is #4361 - @christophd do you think is there anything immediate we can do to include in 1.12.1 by any chance?

**Comment by oscerd** [comment]
Sure

**Comment by squakez** [comment]
Pending fix: https://github.com/apache/camel-k-runtime/pull/1025 - I will test in nightly once this is merged in the runtime to confirm the solution is okey.

**Comment by christophd** [comment]
@squakez regarding #4361 yes, I think we need to apply some fix for 1.12.1

**Comment by squakez** [comment]
> @squakez regarding #4361 yes, I think we need to apply some fix for 1.12.1

Thanks, do you think is it something that can be done during this week? in such case we can postpone the release for some more days.

**Comment by christophd** [comment]
Yes, I have proposed a short term fix in #4361 that we could do. It would be possible to incorporate it this week

**Comment by christophd** [comment]
https://github.com/apache/camel-k/pull/4402

**Comment by squakez** [comment]
> #4402

Okey, let's tentatively have a fix for this during this week before starting the new release process.

**Comment by squakez** [comment]
I think all is sorted out. I will start the release process in the afternoon.

**Comment by squakez** [comment]
I completed the staging process but we have some error:
```
./kamel install --operator-image=camelk/camel-k:1.12.1 --maven-repository=https://repository.apache.org/content/repositories/orgapachecamel-1565
...
camel-k-operator-57df6b67b4-4w4ww camel-k-operator {"level":"info","ts":1685371623.7837243,"logger":"camel-k.controller.build","msg":"Maximum number of running builds (0) exceeded - the build gets enqueued","request-namespace":"default","request-name":"test","max-running-builds-limit":0,"api-version":"camel.apache.org/v1","kind":"Build","ns":"default","name":"kit-chqbk671bknc739bgpig"}
```

**Comment by squakez** [comment]
I amend my previous comment. After cleaning the cluster, all is running as expected. Proceeding to send vote communication.

**Comment by romain-pfund** [comment]
@squakez: What do you mean with "cleaning the cluster". I'm facing the same issue.
 
 I tried `kamel reset` then`kamel uninstall` and install the 1.12.1-nightly release, I get the error when I apply an integration

**Comment by squakez** [comment]
Yeah, I've uninstalled and deleted the IntegrationPlatform as well.

**Comment by romain-pfund** [comment]
sorry for my comment:
seams that
`kamel uninstall --all --global --olm=false`
does the job

**Comment by squakez** [comment]
> sorry for my comment: seams that `kamel uninstall --all --global --olm=false` does the job

Not at all, your comment was more than pertinent, as I double checked and, although we can workaround this by uninstalling the IntegrationPlatform, it still will disrupt any upgrade from older versions to the new version. The problem lays in https://github.com/apache/camel-k/pull/4284 as we have introduced a new parameter that, by default, it is set to 0 when no value is provided. This is happening when we upgrade from a previous IP which is missing the value, and it's initialized to 0 by default.

@christophd would you mind having a look? (I suspect the same applies for version 1.10).

@oscerd, as this is on voting, I guess we need to fix this and then restart a new voting process? Thanks for the clarification

**Comment by oscerd** [comment]
Yes, we need to cancel the vote and do the process again for the camel-k part.

**Comment by christophd** [comment]
@squakez this should be covered already when the operator version changes and the platform is reconciled with that new version. With this the proper values should be set on the platform (https://github.com/apache/camel-k/blob/main/pkg/platform/defaults.go#L370)

do you see some log entry `IntegrationPlatform version updated ...`? does the platform reflect the new operator version in its status?

**Comment by christophd** [comment]
I just did 

- `kamel install --global` with 1.12.0
- `kamel install --global --force` with 1.12.1

The IP now shows the new version 1.12.1 in its status and it has `maxRunningBuilds` property set properly with value > 0

**Comment by christophd** [comment]
Is it possible that you have had a pending/running/incomplete build while updating? The `kamel uninstall --all --global` may have deleted that pending build and after that all is good

**Comment by christophd** [comment]
right now I think this issue is only for existing builds that are already submitted or pending while performing the operator update. these builds will get the `maxRunningBuilds=0` setting as it is a new property in the BuildSpec (https://github.com/apache/camel-k/blob/main/pkg/apis/camel/v1/build_types.go#L50)

**Comment by squakez** [comment]
Steps to reproduce from a clean cluster state:
```
$ kamel install # version 1.12.0
$ kamel install --operator-image=camelk/camel-k:1.12.1 --maven-repository=https://repository.apache.org/content/repositories/orgapachecamel-1565 --olm=false --force # version 1.12.1
$ kamel run /tmp/Test.java 
Integration "test" created
$ stern camel-k-operator
....
camel-k-operator-65d8657fff-4s6hb camel-k-operator {"level":"info","ts":1685624071.3318849,"logger":"camel-k.controller.build","msg":"Maximum number of running builds (0) exceeded - the build gets enqueued","request-namespace":"default","request-name":"test","max-running-builds-limit":0,"api-version":"camel.apache.org/v1","kind":"Build","ns":"default","name":"kit-chs9a1ven07s73abbl80"}
```
Probably the fact you're using a global installation mode is the reason why this is not thrown on your side.

**Comment by squakez** [comment]
BTW, this is not detected by our E2E test, because those tests covering this problem (upgrade checks) are failing after #4312 - one of the main reason we need to release 1.12.1 and start trusting again our CI checks.

**Comment by christophd** [comment]
Sorry, I am not able to reproduce, also with your provided steps. I am using kamel CLI v1.12.0 and v1.12.1 to install the respective operator versions

I can see the platform being updated with the new operator version and I can see the build getting the proper `maxRunningBuilds` setting from that platform

**Comment by squakez** [comment]
I think #4449 can give us some better visibility on the issue. It seems the problem is the usage of the `kamel` CLI which does not contain newly created CRDs.

**Comment by squakez** [comment]
Yay! I managed to fix the check. I will backport to release branches as well, so we can finally have a trusted process.

**Comment by squakez** [comment]
The checks are now failing with something different: likely https://github.com/apache/camel-k/blob/main/e2e/support/test_staging_hooks.go#L34

**Comment by squakez** [comment]
```
camel-k-operator-65d8657fff-jcp2t camel-k-operator {"level":"info","ts":1686130055.6772263,"logger":"camel-k.controller.build","msg":"Maximum number of running builds (0) exceeded - the build gets enqueued","request-namespace":"default","request-name":"test","max-running-builds-limit":0,"api-version":"camel.apache.org/v1","kind":"Build","ns":"default","name":"kit-ci04qbdk390s73d0bn70"}
```
This is still a problem when trying to upgrade from a previous `kamel` cli. Working to find a solution.

**Comment by squakez** [comment]
On vote again. All checks cleared.
